### ISSUE TITLE
BLTouch on Duet2Ethernet board cannot get it near 0.02 on Z_Tilt_adjust

### ISSUE THREAD

**Issue by paulhew69** [KevinOConnor/klipper#3261]
I have spent the last 6 hours trying to get the Z_TILT_ADJUST to actually get to or below 0.02.
Q66 and Coffee on the Klipper Discord have been really helpful and have persevered to assist.

I have a Railcore II, Duet2+Duex, 3 steppers with leadscrews on the bed, MRW kinematics on the bed. No plastic in sight!

I am trying desperately to get away from RRF3.
With RRF3, I send a file to print, it will heat the bed and extruder to 100 / 150, home all axis, then trams the bed / adjusts its tilt, then heats bed / extruder to required temps and prints. There is a macro that does all of this.

So with RRF3 this happens
```Leadscrew adjustments made: -0.011 0.022 0.013, points used 4, (mean, deviation) before (0.010, 0.021) after (-0.000, 0.015)
Bed Traming Cycle: 1 - Difference was 0.021mm
Leadscrew adjustments made: 0.000 0.001 0.004, points used 4, (mean, deviation) before (0.002, 0.015) after (0.000, 0.015)
Bed Traming Cycle: 2 - Difference was 0.015mm```

It is able to achieve 0.015mm in 2 passes
The script is set to retry if the value is above 0.02, max 5 times.

With Klipper there seems to be no consistency, it goes down a little then up a lot and really am trying to get it to work.
I have been through probe_accuracy everything and am getting nowhere, and would like a little input please of what could be going wrong.

One of my Z_TILT_ADJUST results
// Retries: 0/5 Probed points range: 0.030833 tolerance: 0.020000
// Retries: 1/5 Probed points range: 0.033958 tolerance: 0.020000
// Retries: 2/5 Probed points range: 0.029375 tolerance: 0.020000
// Retries: 3/5 Probed points range: 0.038750 tolerance: 0.020000
// Retries: 4/5 Probed points range: 0.038542 tolerance: 0.020000
// Retries: 5/5 Probed points range: 0.030208 tolerance: 0.020000

Probe accuracy testing - this is just some of them.
1st - // probe accuracy results: maximum 3.264375, minimum 3.259375, range 0.005000, average 3.261750, median 3.261250, standard deviation **0.001500**
2nd - // probe accuracy results: maximum 4.124375, minimum 4.122500, range 0.001875, average 4.123312, median 4.123125, standard deviation **0.000687**
3rd - // probe accuracy results: maximum 4.985625, minimum 4.981250, range 0.004375, average 4.983312, median 4.983125, standard deviation **0.001282**

This is the last try I had trying to set Z_TILT
// Retries: 5/5 Probed points range: 0.086042 tolerance: 0.020000
// Retries: 4/5 Probed points range: 0.097083 tolerance: 0.020000
// Retries: 3/5 Probed points range: 0.058958 tolerance: 0.020000
// Retries: 2/5 Probed points range: 0.073333 tolerance: 0.020000
// Retries: 1/5 Probed points range: 0.067708 tolerance: 0.020000

Tried heating the bed and nozzle up to my 'normal' tramming temps, turned off the heaters, ran Z_TILT still no good.
Tried using 'Median' as opposed to average.

I would of thought the values would go progressively down, not up then down etc.


Wiped the firmware, reloaded RRF3, stuck the SD card back in, sent a print and it homed and trammed within 2 passes.

I am out of ideas and not clever enough to know how the backend works, I have an idea in layman's terms how it works but not enough to solve my issue.

I have been using a BLT v3, had to replace the pin today, I bent it the other day as Z_TILT_ADJUST had worked, but I forgot to put another G28 Z in the print start macro before starting to print.
I even put a brand new v3.1 on just incase something was wrong with the other, same issues.

Your input would be greatly appreciated.

Regards,
Paul.


[klippy.log](https://github.com/KevinOConnor/klipper/files/5143390/klippy.log)
<!-- Klipper do something undesirable? YOU MUST ATTACH THE KLIPPER LOG FILE.
See: https://github.com/KevinOConnor/klipper/blob/master/docs/Contact.md -->

**Comment by KevinOConnor** [comment]
The most common reason that Z_TILT doesn't immediately reach the desired level is due to minor errors in motor position or probe offsets.  The common solution to this is to use a `retries` parameter (as you have done).

If you think something went wrong during Z_TILT, then you'll need to attach a log from the session where you ran Z_TILT.

Note that in Klipper the tolerance is the distance between motors (and not the difference in probe positions).  That may or not be different from what RRF reports.  I'd also recommend going to 3 probe points (instead of 4) - as that reduces the chance of a minor bed warp confusing the z correction.

-Kevin
